### ISSUE TITLE
How to synchronize after few iterations (not each iteration)?

### ISSUE THREAD

**Issue by y-rok** [horovod/horovod#1624]
Hello.
I am training large deep learning model such as BERT.
However, I have 2080ti which of memory is just 11GB.
Because of OOM, I should train the model with only small batch size.
The problem is that the smaller batch size, the higher communication cost.
For example, when training the same number of training instances, the communication cost of 4 batch size is 4 times bigger than 32 batch size.
So i want each worker to send their gradients after few iterations (not each iteration).

The below is the scenario i want.

1. Worker calculate their gradients 4 times with 4 batch size  (Note that worker does not update their model in this step)
2. Worker send the gradients calculated from 16 training instances to others.
3. And then update the model of each worker.

This scenario is like just calculating 16 batch at a time.

Is it available to implement this with horovod?

**Comment by tgaddair** [comment]
Hey @y-rok, if you're using PyTorch, then you can do this relatively easily using Horovod's built-in support for local gradient accumulation.  Take a look at the [pytorch_imagenet_resnet50.py](https://github.com/horovod/horovod/blob/master/examples/pytorch_imagenet_resnet50.py#L154) example, particularly the option `backward_passes_per_step`, which controls how many forward passes we accumulate before each backward pass.

If using TensorFlow/Keras, this will be more involved, but doable.  It could also result in a useful PR for Horovod if that's something you might be interested in contributing.

**Comment by y-rok** [comment]
@tgaddair Thanks! I am using tensorflow now but pytorch will be an option. And i will consider contributing that feature in tensorflow! :)
